File: LibraryBackend/Library.Models/Entities/UserReview.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Library.Models.Entities
{
    public class UserReview
    {
        public int Id { get; set; }
        public int ReviewedUserId { get; set; }
        public User ReviewedUser { get; set; }
        public int ReviewerUserId { get; set; }
        public User ReviewerUser { get; set; }
        public int Rating { get; set; }
        public string Comment { get; set; }
        public DateTime ReviewDate { get; set; }
        public bool IsApproved { get; set; }
        public bool IsDenied { get; set; }
    }
}

File: LibraryBackend/UI/elibrary_mobile/lib/models/user_review.dart
import 'package:elibrary_mobile/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'user_review.g.dart';

@JsonSerializable()
class UserReview {
  int? id;
  int? rating;
  String? comment;
  DateTime? reviewDate;
  User? reviewerUser;
  User? reviewedUser;

  UserReview({
    this.id,
    this.rating,
    this.comment,
    this.reviewDate,
    this.reviewerUser,
    this.reviewedUser
  });

  factory UserReview.fromJson(Map<String, dynamic> json) => _$UserReviewFromJson(json);
  Map<String, dynamic> toJson() => _$UserReviewToJson(this);
}